pkg/lso/localvolumeset.go

```diff
@@ -0,0 +1,116 @@
+package lso
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	lsoV1alpha1 "github.com/openshift/local-storage-operator/api/v1alpha1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// LocalVolumeSetBuilder provides a struct for localVolumeSet object from the cluster and a localVolumeSet definition.
+type LocalVolumeSetBuilder struct {
+	// localVolumeSet definition, used to create the localVolumeSet object.
+	Definition *lsoV1alpha1.LocalVolumeSet
+	// Created localVolumeDiscovery object.
+	Object *lsoV1alpha1.LocalVolumeSet
+	// Used in functions that define or mutate localVolumeSet definition. errorMsg is processed
+	// before the localVolumeSet object is created
+	errorMsg string
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+}
+
+// NewLocalVolumeSetBuilder creates new instance of LocalVolumeSetBuilder.
+func NewLocalVolumeSetBuilder(apiClient *clients.Settings, name, nsname string) *LocalVolumeSetBuilder {
+	glog.V(100).Infof("Initializing new %s localVolumeSet structure in %s namespace", name, nsname)
+
+	builder := LocalVolumeSetBuilder{
+		apiClient: apiClient,
+		Definition: &lsoV1alpha1.LocalVolumeSet{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the localVolumeSet is empty")
+
+		builder.errorMsg = "localVolumeSet 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The nsname of the localVolumeSet is empty")
+
+		builder.errorMsg = "localVolumeSet 'nsname' cannot be empty"
+	}
+
+	return &builder
+}
+
+// Discover fetches existing localVolumeSet from cluster.
+func (builder *LocalVolumeSetBuilder) Discover() error {
```

Comment:
Please remove this method.
Add 2 methods instead.
1 - func (builder *LocalVolumeDiscoveryBuilder) Get()
2 - func Pull